services/streammanager/kafka/client/producer.go

```diff
@@ -140,28 +138,45 @@ func headers(msg Message) (headers []kafka.Header) {
 }
 
 func isErrTemporary(err error) bool {
+	if errors.Is(err, context.DeadlineExceeded) {
+		return true
+	}
 	isTransientNetworkError := errors.Is(err, io.ErrUnexpectedEOF) ||
 		errors.Is(err, syscall.ECONNREFUSED) ||
 		errors.Is(err, syscall.ECONNRESET) ||
 		errors.Is(err, syscall.EPIPE)
 	if isTransientNetworkError {
 		return true
 	}
+	var tempError interface{ Temporary() bool }
 	if errors.As(err, &tempError) {
 		return tempError.Temporary()
 	}
+	var timeoutError interface{ Timeout() bool }
+	if errors.As(err, &timeoutError) {
+		return timeoutError.Timeout()
+	}
```

Comment:
we can use `os.IsTimeout(err)` which already does more or less the same thing

Reply:
Yes, it should work. I copied the whole block from [here](https://github.com/segmentio/kafka-go/blob/v0.4.38/error.go#L593) and didn't realize I could use the std library for that instead :+1: